test/test_download.py

```diff
@@ -122,8 +123,10 @@ def print_skipping(reason):
         params['outtmpl'] = tname + '_' + params['outtmpl']
         if is_playlist and 'playlist' not in test_case:
             params.setdefault('extract_flat', 'in_playlist')
-            params.setdefault('playlistend', test_case.get(
-                'playlist_mincount', test_case.get('playlist_count', -2) + 1))
+            params.setdefault('playlistend', max(
+                test_case.get('playlist_mincount') or -1,
+                (test_case.get('playlist_count') or -2) + 1,
+                (test_case.get('playlist_maxcount') or -2) + 1))
```

Comment:
it's ok to use the default param here, and IMO easier to read
```suggestion
            params.setdefault('playlistend', max(
                test_case.get('playlist_mincount', -1),
                test_case.get('playlist_count', -2) + 1,
                test_case.get('playlist_maxcount', -2) + 1))
```